tips/TIP-0026/indexer-rest-api.yaml

```diff
@@ -0,0 +1,854 @@
+          description: The maximum amount of items returned in one call. If there are more items, a cursor to the next page is returned too.
+            The parameter is ignored when pageSize is defined via the cursor parameter.
+        - in: query
+          name: cursor
+          schema:
+            type: string
+          example: 0c78e998f5177834ecb3bae1596d5056af76e487386eecb19727465b4be86a790200.10
+          description: Starts the search from the cursor (confirmationMS+outputId.pageSize).
+      responses:
+        '200':
+          description: "Successful operation."
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/OutputsResponse'
+              examples:
+                Query results in 3 outputs:
+                  $ref: >-
+                    #/components/examples/get-outputs-response-three-example
+                No matching outputIds found:
+                  $ref: >-
+                    #/components/examples/get-outputs-empty-response-example
+                Paging - more items found than it can fit on single page:
+                  $ref: >-
+                    #/components/examples/get-outputs-pagesize2-response-example
+        '400':
+          description: "Unsuccessful operation: indicates that the provided data is invalid."
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/BadRequestResponse'
+        '403':
+          description: "Unsuccessful operation: indicates that the endpoint is not available for public use."
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ForbiddenResponse'
+        '500':
+          description: "Unsuccessful operation: indicates that an unexpected, internal server error happened which prevented the node from fulfilling the request."
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/InternalErrorResponse'
+
+  /api/plugins/indexer/v1/aliases/{aliasId}:
+    get:
+      tags:
+        - aliases
+      summary: Returns the outputId of the current unspent alias output for aliasId.
+      description: Returns the outputId of the current unspent alias output for aliasId.
+      parameters:
+        - in: path
+          name: aliasId
+          schema:
+            type: string
+          example: "0xb098c0ee963575a8d505c299a99ce9c32cd4ecb3"
+          description: Unique identifier of the alias.
+          required: true
+      responses:
+        '200':
+          description: "Successful operation."
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/OutputsResponse'
+              examples:
+                Query results in single output:
+                  $ref: >-
+                    #/components/examples/get-outputs-response-single-example
```

Comment:
Why do we return an array if there can be only a single element?

Reply:
Consistency

Reply:
Although we return 404 if nothing is found instead of an empty list….

I am open to change this, we just thought it might make the parsing logic easier

Reply:
We decided to keep it like this